chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -633,6 +633,26 @@ std::vector<Array> DSplit(const Array& ary, std::vector<int64_t> indices) {
     return Split(ary, std::move(indices), 2);
 }
 
+std::vector<Array> VSplit(const Array& ary, int64_t sections) {
+    if (sections < 1) {
+        throw DimensionError("Number of sections must be larger than 0.");
+    }
+
+    if (ary.ndim() < 2) {
+        throw DimensionError("vsplit only works on arrays of 2 or more dimensions.");
+    }
+
+    return Split(ary, sections, 0);
+}
+
+std::vector<Array> VSplit(const Array& ary, std::vector<int64_t> indices) {
+    if (ary.ndim() < 2) {
+        throw DimensionError("vsplit only works on arrays of 3 or more dimensions.");
```

Comment:
Ditto. The error message seems inconsistent with the logic by the way so it'd be good if you could fix it simultaneously.